sdk/metric/meter_test.go

```diff
@@ -64,3 +72,452 @@ func TestMeterRegistry(t *testing.T) {
 		assert.Equal(t, 1, i, "iteration not stopped after first flase return")
 	})
 }
+
+// A meter should be able to make instruments concurrently.
+func TestMeterInstrumentConcurrency(t *testing.T) {
+	wg := &sync.WaitGroup{}
+	wg.Add(6)
+
+	m := NewMeterProvider().Meter("inst-concurrency")
+
+	go func() {
+		_, _ = m.AsyncFloat64().Counter("AFCounter")
+		wg.Done()
+	}()
+	go func() {
+		_, _ = m.AsyncFloat64().UpDownCounter("AFUpDownCounter")
+		wg.Done()
+	}()
+	go func() {
+		_, _ = m.AsyncFloat64().Gauge("AFGauge")
+		wg.Done()
+	}()
+	go func() {
+		_, _ = m.AsyncInt64().Counter("AICounter")
+		wg.Done()
+	}()
+	go func() {
+		_, _ = m.AsyncInt64().UpDownCounter("AIUpDownCounter")
+		wg.Done()
+	}()
+	go func() {
+		_, _ = m.AsyncInt64().Gauge("AIGauge")
+		wg.Done()
+	}()
+
+	wg.Wait()
+}
+
+// A Meter Should be able register Callbacks Concurrently.
+func TestMeterCallbackCreationConcurrency(t *testing.T) {
+	wg := &sync.WaitGroup{}
+	wg.Add(2)
+
+	m := NewMeterProvider().Meter("callback-concurrency")
+
+	go func() {
+		_ = m.RegisterCallback([]instrument.Asynchronous{}, func(ctx context.Context) {})
+		wg.Done()
+	}()
+	go func() {
+		_ = m.RegisterCallback([]instrument.Asynchronous{}, func(ctx context.Context) {})
+		wg.Done()
+	}()
+}
+
+// Instruments should produce correct ResourceMetrics
+// TODO (2814): include sync instruments.
+func TestMeterCreatesInstruments(t *testing.T) {
+	testCases := []struct {
+		name string
+		fn   func(*testing.T, metric.Meter)
+		want metricdata.Metrics
+	}{
+		{
+			name: "Aync Int Count",
```

Comment:
```suggestion
			name: "Async Int Count",
```

Reply:
Or even better so the name doesn't become `Async_Int_Count` in the test output:

```suggestion
			name: "AyncInt64Count",
```

Reply:
This should be done for the other test cases as well.